### ISSUE TITLE
Support for Type Info

### ISSUE THREAD

**Issue by lerouxrgd** [LaurentMazare/tch-rs#604]
Hello,

In PyTorch there are various [type info](https://pytorch.org/docs/stable/type_info.html) such as min/max/etc for each `torch.dtype` available through `torch.finfo` and `torch.iinfo`. 

Such information is used particularly in HuggingFace transformers for [attention masks](https://github.com/huggingface/transformers/commit/ece762443e4269e50cc1a348838e63e4a499575d). While implementing [GPT-J](https://huggingface.co/docs/transformers/model_doc/gptj) for `rust-bert` (see related [PR](https://github.com/guillaume-be/rust-bert/pull/288)) I ran into such [Python code](https://github.com/huggingface/transformers/blob/31d452c68b34c2567b62924ee0df40a83cbc52d5/src/transformers/models/gptj/modeling_gptj.py#L612-L613):

```python
attention_mask = (1.0 - attention_mask) * torch.finfo(self.dtype).min
```

Where `self.dtype` would essentially be `float32` or `float16`. 

I think that the way to handle it in `tch-rs` is to add the type info as const values in the [Element](https://docs.rs/tch/latest/tch/kind/trait.Element.html) trait, do you think this is doable/correct ?

Besides, for the above Python to be ported to Rust it would also be necessary to have `impl From<half::f16> for Scalar`, would this also be doable ?

**Comment by lerouxrgd** [comment]
I am working on [an implementation](https://github.com/LaurentMazare/tch-rs/compare/main...lerouxrgd:tch-rs:type-info?expand=1) for that, any feedback you may have is very welcome.

Now I am realizing that I don't really know what to do for `Kind` values that don't implement `Scalar` (The `todo!()` in the code), do you have any idea/recommendation ?
